### PR TITLE
Parse all files as 'objects', not metadata.

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

This commit also ensures that errors are emitted to stderr, so that they don't get interspersed with the YAML output (on stdout), and avoids the need to type assert objects to *unstructured.Unstructured.

### How has this code been tested
I've run it against the example directory, and ensured that YAML files without an `apiVersion` and `kind` result in an error.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: cmd/pkg-unpack/main.go [hunk 1]
```diff
@@ -48,32 +47,35 @@ func Run(ctx context.Context) error {
 		ObjectTyper:   unstructuredscheme.NewUnstructuredObjectTyper(),
 		ObjectCreater: unstructuredscheme.NewUnstructuredCreator(),
 	}
-	p := parser.New(uScheme, uScheme)
+
+	// We want to leverage the package parser, but we don't need to make a
+	// distinction between package metadata and payload objects. Passing en
```
Comment (hasheddan) [inline cmd/pkg-unpack/main.go]: nitpick:
```suggestion
	// distinction between package metadata and payload objects. Passing in
```

File: cmd/pkg-unpack/main.go [hunk 2]
```diff
@@ -48,32 +47,35 @@ func Run(ctx context.Context) error {
...
+	// empty 'metadata' scheme and an unstructured 'object' scheme will cause
+	// the parser to treat any valid Kubernetes YAML (including package metadata
+	// objects) to be treated as regular payload objects. Any object that
```
Comment (hasheddan) [inline cmd/pkg-unpack/main.go]: Looks like comment got cut off

**Review by hasheddan — COMMENTED** [review]
Just a few minor nitpicks, otherwise LGTM

File: cmd/pkg-unpack/main.go (hunk 2 above)
Comment (negz) [reply cmd/pkg-unpack/main.go]: The dangers of trying to code and meet at the same time. Fixed.

**Review by hasheddan — APPROVED** [review]
LGTM!

**Review by muvaf — APPROVED** [review]

**Pull request merged**